libs/util/git/src/git-client.service.ts

```diff
@@ -282,29 +281,41 @@ export class GitClientService {
       baseBranch,
     });
 
-    const { diff } = await this.preCommitProcess({
+    // calculate diff on everything *already existing* on the amplication branch since last amplication build
+    const preCommitDiff = await this.preCommitProcess({
       branchName,
       gitCli,
+      useBeforeLastCommit: false,
     });
 
+    // now commit the new build
     const sha = await gitCli.commit(branchName, commitMessage, preparedFiles);
     this.logger.debug("New commit added", { sha });
 
-    if (diff) {
-      const diffFolder = normalize(
-        join(
-          `.amplication/diffs`,
-          this.provider.name,
-          owner,
-          repositoryName,
-          v4()
-        )
+    // now calculate the diff (only) between the previous amplication build and the new one
+    const postCommitDiff = await this.preCommitProcess({
+      branchName,
+      gitCli,
+      useBeforeLastCommit: true,
+    });
+
+    // On top of the previous amplication build, apply the patch for the already existing changes that have been in this branch
+    if (preCommitDiff.diff) {
+      await this.applyPostCommit(
+        preCommitDiff.diff,
+        owner,
+        repositoryName,
+        gitCli
+      );
+    }
+    // on top of the already existing changes, apply the diff between the previous amplication build and the new one
+    if (postCommitDiff.diff) {
```

Comment:
How can this not wipe the conflict resolution chosen committed by the user before the new build?

Reply:
I couldn't reproduce an example of how this would happen, if you have a scenario in mind, please test it with this branch and see if you can create the problem you are describing in reality.

Reply:
We will test it properly on sandbox with a simple repo